ground/src/main/java/com/google/android/ground/ui/map/Feature.kt

```diff
@@ -17,11 +17,28 @@ package com.google.android.ground.ui.map
 
 import com.google.android.ground.model.geometry.Geometry
 
-/** Represents an individual feature on a map with a given [Geometry]. */
-data class Feature(val id: String, val tag: Type, val geometry: Geometry) {
-  /** Denotes the kind of entity this map feature represents. */
+/** Represents an individual feature on a map with a given [Geometry] and [Tag]. */
+data class Feature(val tag: Tag, val geometry: Geometry) {
+  /**
+   * Denotes the kind of entity this map feature represents and contains any additional data it
+   * carries.
+   */
+  data class Tag(
+    /** A unique identifier for the model object that this feature represents. */
+    val id: String = "",
```

Comment:
Why does it have a default value? Shouldn't this always be non-null & non-empty?

Reply:
My thinking was that there may have been cases in which we don't need an ID right away (e.g. user-drawn polygons), but you're right that we probably do just need one in all cases.